include/mbedtls/ssl_ciphersuites.h

```diff
@@ -371,10 +371,18 @@ typedef enum {
 #endif
 
 /* TLS 1.2 or 1.3 key exchanges using ECDH(E) */
+#if defined(MBEDTLS_KEY_EXCHANGE_SOME_ECDH_OR_ECDHE_1_2_ENABLED) || \
+    (defined(MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_EPHEMERAL_ENABLED) || \
+    defined(MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_EPHEMERAL_ENABLED)) && \
+    !defined(PSA_WANT_ALG_FFDH)
```

Comment:
I don't think that's correct, that should be `defined(PSA_WANT_ALG_ECDH)` (we can have one of them or both, so !FFDH implies ECDH but they're not equivalent).

Also, we're missing a level of parentheses. We can simplify this by using existing macros, and perhaps also defining an intermediate macro for clarity:
```c
#if defined(MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_SOME_EPHEMERAL_ENABLED) && \
    defined(PSA_WANT_ALG_ECDH)
#define MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_SOME_ECDHE_ENABLED
#endif

#if defined(MBEDTLS_KEY_EXCHANGE_SOME_ECDH_OR_ECDHE_1_2_ENABLED) || \
    defined(MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_SOME_ECDHE_ENABLED)
#define MBEDTLS_KEY_EXCHANGE_SOME_ECDH_OR_ECDHE_ANY_ENABLED
#endif
```

Reply:
Right, I tried to shortcut with `!PSA_WANT_ALG_FFDH` but apparently I missed something :(